synapse/federation/transport/server/_base.py

```diff
@@ -140,7 +145,7 @@ def _parse_auth_header(header_bytes):
         header_bytes (bytes): header value
 
     Returns:
-        Tuple[str, str, str]: origin, key id, signature.
+        Tuple[str, str, str, Optional[str]]: origin, key id, signature, destination.
```

Comment:
While we're here, it'd be great if these types could be added as annotations to the function itself (and removed from the docstring)

Reply:
I'd say that this might be out of scope of this PR, but if you want I can include that. wdyt?

Reply:
I don't mind leaving it be for now; at some point we'll have to do a pass over this file to improve type coverage anyway.